video_xblock/static/js/download-handout.js

```diff
@@ -0,0 +1,207 @@
+
+        $resetButton.click(function() {
+            var defaults = JSON.parse($wrapper.attr('data-default'));
+            $checkboxes.each(function() {
+                var val = JSON.parse($(this).val());
+                $(this).prop('checked', defaults.indexOf(val) > -1);
+            });
+            $wrapper.removeClass('is-set');
+            $resetButton.removeClass('active').addClass('inactive');
+        });
+    });
+
+    var studio_submit = function(data) {
+        var handlerUrl = runtime.handlerUrl(element, 'submit_studio_edits');
+        runtime.notify('save', {state: 'start', message: gettext("Saving")});
+        $.ajax({
+            type: "POST",
+            url: handlerUrl,
+            data: JSON.stringify(data),
+            dataType: "json",
+            global: false,  // Disable Studio's error handling that conflicts with studio's notify('save') and notify('cancel') :-/
+            success: function(response) { runtime.notify('save', {state: 'end'}); }
+        }).fail(function(jqXHR) {
+            var message = gettext("This may be happening because of an error with our server or your internet connection. Try refreshing the page or making sure you are online.");
+            if (jqXHR.responseText) { // Is there a more specific error message we can show?
+                try {
+                    message = JSON.parse(jqXHR.responseText).error;
+                    if (typeof message === "object" && message.messages) {
+                        // e.g. {"error": {"messages": [{"text": "Unknown user 'bob'!", "type": "error"}, ...]}} etc.
+                        message = $.map(message.messages, function(msg) { return msg.text; }).join(", ");
+                    }
+                } catch (error) { message = jqXHR.responseText.substr(0, 300); }
+            }
+            runtime.notify('error', {title: gettext("Unable to update settings"), message: message});
+        });
+    };
+
+    $('.save-button', element).bind('click', function(e) {
+        e.preventDefault();
+        var values = {};
+        var notSet = []; // List of field names that should be set to default values
+        for (var i in fields) {
+            var field = fields[i];
+            if (field.isSet()) {
+                values[field.name] = field.val();
+            } else {
+                notSet.push(field.name);
+            }
+            // Remove TinyMCE instances to make sure jQuery does not try to access stale instances
+            // when loading editor for another block:
+            if (field.hasEditor()) {
+                field.removeEditor();
+            }
+        }
+        studio_submit({values: values, defaults: notSet});
+        var filehandlerUrl = runtime.handlerUrl(element, 'save_file');
+        $.ajax({
+           url: filehandlerUrl,
+            type: "POST",
+            data: JSON.stringify({file: uploadFile})
+        });
+    });
+
+    var $fileUploader = $('.input-file-uploader', element);
```

Comment:
could you add some comments about the places where you extended default templates, so in the future it might be simplier upgrade this file.